spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -459,14 +459,16 @@ def main(argv: Sequence[str]):
                         list_data = get_data_arrays(data.shape, arg_value)
                     except ValueError as e:
                         printv(f"ERROR: -{arg_name}: {e}", 1, 'error')
-                    data += np.sum(list_data, axis=0)
+                    # the addition can't use += because the dtypes could be different
+                    data = data + np.sum(list_data, axis=0)
```

Comment:
The error message mentions `casting='same_kind'`, but I'm struggling to figure out how `v1 += v2` and `v1 = v1 + v2` map to numpy's 5 casting rules ([`{‘no’, ‘equiv’, ‘safe’, ‘same_kind’, ‘unsafe’}`](https://numpy.org/devdocs/reference/generated/numpy.can_cast.html)). It's hard to find information on the underlying mechanics?

But, maybe I'm overthinking things by trying to understand the exact mechanics. A quick test shows that... practically, this change fixes the issue, and `ints` will be promoted to `floats`.

Reply:
I think you can look at `np.ufunc.resolve_dtypes` to see the difference between regular addition, and in-place addition.

For regular addition (the triple of dtypes is (left input, right input, output), and `None` means "unconstrained"):
```py
>>> np.add.resolve_dtypes((np.dtype('int8'), np.dtype('float64'), None))
(dtype('float64'), dtype('float64'), dtype('float64'))
```
I'm interpreting this to mean:
- The left input gets cast from `int8` to `float64`.
- The right input stays `float64`.
- The output (which is a newly allocated array) will be `float64`.

For in-place addition (meaning, it tries to re-use/modify the existing allocation of the left input for the output):
```py
>>> np.add.resolve_dtypes((np.dtype('int8'), np.dtype('float64'), np.dtype('int8')))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'add' output from dtype('float64') to dtype('int8') with casting rule 'same_kind'
```
I'm interpreting this to mean:
- The output should be a `float64` array, but this can't be stored into an existing `int8` array.